test/test_tensor_creation_ops.py

```diff
@@ -3382,6 +3382,14 @@ def test_normal_std_error(self, device):
             with self.assertRaisesRegex(RuntimeError, r'normal expects all elements of std >= 0.0'):
                 torch.normal(input, std)
 
+    def test_normal_default_device(self, device):
+        try:
+            torch.set_default_device(device)
+            t = torch.normal(0, 1, (10, 10))
+        finally:
+            torch.set_default_device(None)
+        self.assertEqual(str(t.device), device)
```

Comment:
Shouldn't you set back the previous default device instead?

Like 
```suggestion
        default_device = torch.default_device()
        try:
            torch.set_default_device(device)
            t = torch.normal(0, 1, (10, 10))
        finally:
            torch.set_default_device(default_device)
        self.assertEqual(str(t.device), device)
```

Also, note that this works
```python
torch.normal(0, 1, (10, 10))
```
but this doesn't
```python
torch.normal(0, 1)
```